test/WebSites/RazorPagesWebSite/Pages/ShortCircuitPage.cshtml.cs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace RazorPagesWebSite.Pages
+{
+    [AsyncTestFiltersAttributeAttribute]
+    [SyncTestFiltersAttributeAttribute]
+    public class ShortCircuitPageModel : PageModel
+    {
+        public IActionResult OnGet()
+        {
+            return Page();
+        }
+
+        private static bool ShouldShortCircuit(HttpContext httpContext, string currentTargetName)
```

Comment:
Let's split this out in to two pages. One for resource and one for page filter. This is kinda gross